var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -4,205 +4,133 @@
-            gcc_base = os.path.split(os.path.split(self.compiler.cc)[0])[0]
-            env.prepend_path('LD_LIBRARY_PATH', join_path(gcc_base, 'lib64'))
-
-    def install(self, spec, prefix):
-        # Julia needs git tags
-        if os.path.isfile('.git/shallow'):
-            git = which('git')
-            git('fetch', '--unshallow')
-        # Explicitly setting CC, CXX, or FC breaks building libuv, one
-        # of Julia's dependencies. This might be a Darwin-specific
-        # problem. Given how Spack sets up compilers, Julia should
-        # still use Spack's compilers, even if we don't specify them
-        # explicitly. Potential options are
-        # 'CC=cc',
-        # 'CXX=c++',
-        # 'FC=fc',
-        # 'USE_SYSTEM_ARPACK=1',
-        # 'override USE_SYSTEM_CURL=1',
-        # 'USE_SYSTEM_FFTW=1',
-        # 'USE_SYSTEM_GMP=1',
-        # 'USE_SYSTEM_MPFR=1',
-        # 'USE_SYSTEM_PCRE=1',
+        # this is a bit ridiculous, but we are setting runtime linker paths to
+        # dependencies so that libwhich can locate them.
+        pkgs = [
+            'curl', 'dsfmt', 'gmp', 'libgit2', 'libssh2', 'libunwind', 'mbedtls',
+            'mpfr', 'nghttp2', 'openblas', 'openlibm', 'pcre2', 'suite-sparse',
+            'utf8proc', 'zlib', 'libblastrampoline'
+        ]
+        for pkg in pkgs:
+            for dir in self.spec[pkg].libs.directories:
+                env.prepend_path('LD_LIBRARY_PATH', dir)
+                env.prepend_path('DYLD_FALLBACK_LIBRARY_PATH', dir)
+
+    def edit(self, spec, prefix):
         options = [
-            'prefix={0}'.format(prefix)
+            'prefix={0}'.format(prefix),
+            'USE_BINARYBUILDER=0',
+
+            # Spack managed dependencies
+            'USE_SYSTEM_BLAS=1',
+            'USE_SYSTEM_CSL=1',
+            'USE_SYSTEM_CURL=1',
+            'USE_SYSTEM_DSFMT=1',
+            'USE_SYSTEM_GMP=1',
+            'USE_SYSTEM_LIBBLASTRAMPOLINE=1',
+            'USE_SYSTEM_LIBGIT2=1',
+            'USE_SYSTEM_LIBSSH2=1',
+            'USE_SYSTEM_LIBSUITESPARSE=1',
+            'USE_SYSTEM_LIBUNWIND=1',
+            'USE_SYSTEM_LIBUV=1',
+            'USE_SYSTEM_LIBWHICH=1',
+            'USE_SYSTEM_LLVM=1',
+            'USE_SYSTEM_MBEDTLS=1',
+            'USE_SYSTEM_MPFR=1',
+            'USE_SYSTEM_P7ZIP=1',
+            'USE_SYSTEM_PATCHELF=1',
+            'USE_SYSTEM_PCRE=1',
+            'USE_SYSTEM_UTF8PROC=1',
+            'USE_SYSTEM_ZLIB=1',
+
+            # todo: ilp depends on arch; prefix depends on ilp
+            'USE_BLAS64=1',
+            'LIBBLASNAME=libopenblas64_',
+            'LIBLAPACKNAME=libopenblas64_',
+            'USE_SYSTEM_LAPACK=1',
+            'LIBUV={0}'.format(spec['libuv'].libs.libraries[0]),
+            'LIBUV_INC={0}'.format(spec['libuv'].headers.directories[0]),
+            'USE_LLVM_SHLIB=1',
+            # make rebuilds a bit faster for now, not sure if this should be kept
+            'JULIA_PRECOMPILE={0}'.format(
```

Comment:
Probably not. The risk of someone shooting themselves in the foot is huge xD

Reply:
Yeah, I still have it under a spack variant +/~precompile and it's turned on by default, it can still be useful for developers